### ISSUE TITLE
ssh-add -l shows the key in the agent but ssh doesn't pick it up 

### ISSUE THREAD

**Issue by chx** [PowerShell/Win32-OpenSSH#706]
Please answer the following

**"OpenSSH for Windows" version**
0.0.12.0

**OS details**
 Windows 10 Pro

**What is failing**
ssh asks for the password of my key despite ssh-add -l shows the key

**Comment by manojampalam** [comment]
It most likely means that ssh has tried to authenticated using that key, server rejected it and ssh has fallen back to password based auth.

**Comment by chx** [comment]
It is asking for the password of the key and it uses the key properly if the password is given.

**Comment by manojampalam** [comment]
Ok. sorry, I misread your first post. Can you dump the client side logs using ssh -v ?

**Comment by chx** [comment]
There's some mismatch here directory wise as well because it doesn't find the .ssh dir in the homedir. The whole thing is strange -- but I have not ran Windows in 13 years just now the Linux for Windows userspace made me switch back to Windows after all these years and everything is strange.

	PS C:\Users\chx> ssh-add -l
	4096 SHA256:Zq4rPrdgEYpZgHrE6oRpNPwR37dMy75/GhGs32onn2k C:\Users\chx/.ssh/id_rsa (RSA)
	PS C:\Users\chx> ssh -v chx@github.com
	OpenSSH_7.1p1, OpenSSL 1.0.2e 3 Dec 2015
	debug1: Connecting to github.com [192.30.255.112] port 22.
	debug1: Connection established.
	Could not create directory '/home/chx/.ssh'.
	debug1: key_load_public: No such file or directory
	debug1: identity file /home/chx/.ssh/id_rsa type -1
	debug1: key_load_public: No such file or directory
	debug1: identity file /home/chx/.ssh/id_rsa-cert type -1
	debug1: key_load_public: No such file or directory
	debug1: identity file /home/chx/.ssh/id_dsa type -1
	debug1: key_load_public: No such file or directory
	debug1: identity file /home/chx/.ssh/id_dsa-cert type -1
	debug1: key_load_public: No such file or directory
	debug1: identity file /home/chx/.ssh/id_ecdsa type -1
	debug1: key_load_public: No such file or directory
	debug1: identity file /home/chx/.ssh/id_ecdsa-cert type -1
	debug1: key_load_public: No such file or directory
	debug1: identity file /home/chx/.ssh/id_ed25519 type -1
	debug1: key_load_public: No such file or directory
	debug1: identity file /home/chx/.ssh/id_ed25519-cert type -1
	debug1: Enabling compatibility mode for protocol 2.0
	debug1: Local version string SSH-2.0-OpenSSH_7.1
	debug1: Remote protocol version 2.0, remote software version libssh-0.7.0
	debug1: no match: libssh-0.7.0
	debug1: Authenticating to github.com:22 as 'chx'
	debug1: SSH2_MSG_KEXINIT sent
	debug1: SSH2_MSG_KEXINIT received
	debug1: kex: server->client chacha20-poly1305@openssh.com <implicit> none
	debug1: kex: client->server chacha20-poly1305@openssh.com <implicit> none
	debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
	debug1: Server host key: ssh-rsa SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8
	The authenticity of host 'github.com (192.30.255.112)' can't be established.
	RSA key fingerprint is SHA256:nThbg6kXUpJWGl7E1IGOCspRomTxdCARLviKw6E5SY8.
	Are you sure you want to continue connecting (yes/no)? yes
	Failed to add the host to the list of known hosts (/home/chx/.ssh/known_hosts).
	debug1: SSH2_MSG_NEWKEYS sent
	debug1: expecting SSH2_MSG_NEWKEYS
	debug1: SSH2_MSG_NEWKEYS received
	debug1: Roaming not allowed by server
	debug1: SSH2_MSG_SERVICE_REQUEST sent
	debug1: SSH2_MSG_SERVICE_ACCEPT received
	debug1: Authentications that can continue: publickey
	debug1: Next authentication method: publickey
	debug1: Trying private key: /home/chx/.ssh/id_rsa
	debug1: Trying private key: /home/chx/.ssh/id_dsa
	debug1: Trying private key: /home/chx/.ssh/id_ecdsa
	debug1: Trying private key: /home/chx/.ssh/id_ed25519
	debug1: No more authentication methods to try.
	Permission denied (publickey).
	PS C:\Users\chx> ls C:\Users\chx\.ssh\


		Directory: C:\Users\chx\.ssh


	Mode                LastWriteTime         Length Name
	----                -------------         ------ ----
	-a----       2016-10-26   2:02 PM           1261 authorized_keys
	-a----       2016-09-06   8:56 AM            664 config
	-a----       2009-11-23  10:11 AM           1697 identity
	-a----       2009-09-09   6:48 AM            736 id_dsa
	-a----       2009-09-09   6:48 AM            602 id_dsa.pub
	-a----       2015-08-15  10:13 AM           3326 id_rsa
	-a----       2009-09-09   6:48 AM           1675 id_rsa.old
	-a----       2015-08-15  10:13 AM            737 id_rsa.pub
	-a----       2017-04-27  10:03 AM          86722 known_hosts
	-a----       2017-04-24  10:12 PM          86722 known_hosts.old
	-a----       2012-11-01   6:51 PM            672 servers.privatekey

**Comment by manojampalam** [comment]
This is definitely not the client coming from this project:
OpenSSH_7.1p1, OpenSSL 1.0.2e 3 Dec 2015

Can you check where its getting picked from? do this
where ssh.exe
Also check where ssh-add is getting picked from?

**Comment by chx** [comment]
I installed https://chocolatey.org/packages/openssh .First without the server feature because you had a bug where ssh-agent wasn't working if the server wasn't installed and then with the server feature and apparently this confused the poor thing. There was an `ssh.exe` under chocolatey\bin and one under `C:\Program Files\OpenSSH-Win64` where the server lives.

This will likely not be a problem because I saw the other issue is now closed and the next release will work sshd-less.

Note: `where` does not seem to work under Powershell. `where.exe` does.

One more note: if you install `git` with chocolatey as I did -- it also contains an ssh.exe which doesn't want to talk to this ssh-agent, just rename `C:\Program Files\git\usr\bin\ssh.exe` to say `_ssh.exe` or delete it outright and all will work.

Thanks for the kind and speedy assistance.
